pkg/kademlia/kademlia.go

```diff
@@ -329,7 +356,13 @@ func (k *Kad) Disconnected(addr swarm.Address) {
 	k.connectedPeers.Remove(addr, po)
 
 	k.waitNextMu.Lock()
-	k.waitNext[addr.String()] = time.Now().Add(timeToRetry)
+	failedAttempts := 0
+	info, ok := k.waitNext[addr.String()]
+	if ok {
+		failedAttempts = info.failedAttempts + 1
```

Comment:
I would assume that fieldAttempts would be incremented only if the connection fails, not on every disconnect, as disconnect may be intentional by the peer. Also reseting failedAttempts to 0 should be on successful connection, I think.

Reply:
OMG, good catch, I was debugging and changing this and forgot to change back 🙈  thanks!